lib/private/docker.ts

```diff
@@ -194,13 +205,12 @@ export class Docker {
     this.configDir = undefined;
   }
 
-  private async execute(args: string[], options: ShellOptions = {}) {
+  private async execute(args: string[], options: ShellOptions) {
```

Comment:
To go with my previous comment:

```suggestion
  private async execute(args: string[], options: Omit<ShellOptions, 'shellEventPublisher'>) {
```

To replace the `logger: this.logger` that you removed.

Reply:
I have never seen that before in my life

Reply:
It needed to be Exclude, not Omit

Reply:
Have you looked it up and are you up to speed on what it does now? 

Otherwise, read this: https://www.typescriptlang.org/docs/handbook/2/mapped-types.html

Reply:
In addition to saying "I've never seen this before", please confirm whether you understand what I'm trying to say here, or indicate that you're going to ignore my suggestion (with motivation please).

Reply:
Yes sorry you did a little bit of a pre-emptive review here but I should have let you know it wasn't ready before I left, I just hadn't had a chance to address this. I was merely remarking it's new syntax to me but I certainly was able to surmise what it was doing when I read it, pretty cool.